garage/logger/snapshotter.py

```diff
@@ -0,0 +1,90 @@
+"""Allows the taking of snapshots."""
+from os import path as osp
+import pickle
+
+
+class Snapshotter:
+    """This class handles the creation of snapshots."""
+
+    def __init__(self):
+        self._snapshot_dir = None
+        self._snapshot_mode = 'all'
+        self._snapshot_gap = 1
+
+    @property
+    def snapshot_dir(self, ):
```

Comment:
hmm does this really need to be changed throughout the lifecycle of a snapshotter? why not just force users to make a new Snapshotter for a new directory? Then we can put this logic in the constructor